cxx4opencl/diff2openclc.txt

```diff
@@ -0,0 +1,121 @@
+// Copyright 2019-2020 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+=== Difference to OpenCL C
+
+{cpp} for OpenCL aims to achieve backwards compatibility with OpenCL C
+for the majority of features. However, it is a different
+language that is derived from {cpp} and therefore it inherits
+its fundamental design principles. Hence {cpp} for OpenCL
+deviates from OpenCL C in the same areas where {cpp} deviates
+from C.
+
+This section describes the main differences between {cpp} for
+OpenCL and OpenCL C.
+
+==== Implicit Conversions
+
+{cpp} is much stricter about conversions between types,
+especially those that are performed implicitly by the compiler.
+For example it is not possible to convert a `const` object
+to non-`const` implicitly. For details please refer to {cpp}17
+(`ISO/IEC 14882:2017`) `[conv]`.
+
+[source,c]
+----------
+void foo(){
+  const int *ptrconst;
+  int *ptr = ptrconst; // invalid initialization discards const qualifier
+}
+----------
+
+The same applies to narrowing conversions in
+initialization lists ({cpp}17 `[dcl.init.list]`). 
+
+[source,c]
+----------
+struct mytype {
+ int i;
+};
+void foo(uint par){
+  mytype var = {
+    .i = par // narrowing from uint to int is disallowed
+  };
+}
+----------
```

Comment:
This example should be simplified. The structure isn't really needed.

Reply:
Hmm... I don't know how to show it without... Can you paste an example you have in mind?

Reply:
Oh, I'd read a bit too quickly. The restriction only applies to initialisation lists. The example is fine as is.